test/index.test.ts

```diff
@@ -1,89 +1,79 @@
-      const request = constructRequest(harExamples.full);
+  let fetchHAR;
 
-      expect(request.url).to.equal('https://httpbin.org/post?key=value&foo=bar&foo=baz&baz=abc');
-      expect(request.method).to.equal('POST');
+  beforeEach(function () {
+    /**
+     * Under Node 18's native `fetch` implementation if a `File` global doesn't exist it'll polyfill
+     * its own implementation. Normally this works fine, but its implementation is **different**
+     * than the one that `formdata-node` ships and when we use the `formdata-node` one under Node
+     * 18 `type` options that we set into `File` instances don't get picked up, resulting in
+     * multipart payloads being sent as `application/octet-stream` instead of whatever content type
+     * was attached to that file.
+     *
+     * This behavior also extends to Undici's usage of `Blob` as well where the `Blob` that ships
+     * with `formdata-node` behaves differently than the `Blob` that is part of the Node `buffer`
+     * module, which Undici wants you to use.
+     *
+     * `NODE_ENV` is set to `production` when Karma runs this code and we don't need to polyfill
+     * APIs that are already available in the browser.
+     */
+    if (process.env.NODE_ENV !== 'production') {
+      if (!globalThis.File) {
+        globalThis.FormData = require('formdata-node').FormData;
+      }
 
-      if (host.node) {
-        expect(Array.from(request.headers)).to.deep.equal([
-          ['accept', 'application/json'],
-          ['content-type', 'application/x-www-form-urlencoded'],
-          ['cookie', 'foo=bar; bar=baz'],
-        ]);
+      const isNode18 = (host.node as VersionInfo).version >= 18;
+      if (isNode18) {
+        globalThis.File = require('undici').File;
+        globalThis.Blob = require('buffer').Blob;
       } else {
-        expect(Array.from(request.headers)).to.deep.equal([
-          ['accept', 'application/json'],
-          ['content-type', 'application/x-www-form-urlencoded'],
-        ]);
+        globalThis.File = require('formdata-node').File;
+        globalThis.Blob = require('formdata-node').Blob;
       }
+    }
 
-      // Inspecting `Request.body` isn't supported in most browsers right now.
-      // https://developer.mozilla.org/en-US/docs/Web/API/Request/body
-      if (host.node) {
-        // The `Request` object in Node 18 has `body` returning a `ReadableStream` (which is
-        // spec-compliant), whereas `node-fetch` returns it as a `Buffer`. Accordingly,
-        // `ReadableStream` was only introduced in Node 16 so if it doesn't exist here then we also
-        // don't want to fail.
-        if (typeof ReadableStream !== 'undefined' && request.body instanceof ReadableStream) {
-          const res = await new Response(request.body).text();
-          expect(res).to.equal('foo=bar');
-        } else {
-          expect(request.body.toString()).to.equal('foo=bar');
-        }
-      }
-    });
+    fetchHAR = require('../src').default;
```

Comment:
We need to load the library here instead of at the top because if we load it at the top, with the polyfill work above above it, Mocha will always load that when it processes tests (even if we're doing a `it.only` in another file).

This is/was leading to issues where tests would pass when run on their own but if you ran the entire suite the `File` and `Blob` polyfills wouldn't get properly applied to Node tests because the `browser-quirks` test runs first, and that was loading fetchHAR into memory. And once it's in memory it's never going to use any polyfill we set into the global state.

Reinstantiating and reloading the library within a `beforeEach` ensures that we always get a repeatable state.

✨  javascript ✨